src/find/matchers/printf.rs

```diff
@@ -619,6 +617,16 @@ impl Matcher for Printf {
                 },
             }
         }
+    }
+}
+
+impl Matcher for Printf {
+    fn matches(&self, file_info: &WalkEntry, matcher_io: &mut MatcherIO) -> bool {
+        if let Some(file) = &self.output_file {
+            self.print(file_info, file);
+        } else {
+            self.print(file_info, &mut *matcher_io.deps.get_output().borrow_mut());
+        }
```

Comment:
a bit more rust idiomatic:

```suggestion
      let output = self.output_file.as_ref()
        .unwrap_or_else(|| matcher_io.deps.get_output().borrow_mut());
    self.print(file_info, output);
```

Reply:
Wow, that's much cleaner.

I tried modifying it but I seem to be having some type issues.

```
error[E0308]: mismatched types
   --> src/find/matchers/printf.rs:628:32
    |
628 |             .unwrap_or_else(|| matcher_io.deps.get_output().borrow_mut());
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
                                     expected `&File`, found `RefMut<'_, dyn Write>`
    |
    = note: expected reference `&File`
                  found struct `RefMut<'_, (dyn std::io::Write + 'static)>`
```

and I also tried:
```
fn print(&self, file_info: &WalkEntry, out: &mut dyn Write) { ... }

...
let output: &mut dyn Write = self.output_file
    .as_ref()
    .map(|file| file.borrow_mut() as &mut dyn Write)
    .unwrap_or_else(|| matcher_io.deps.get_output().borrow_mut());

self.print(file_info, output);
...
```

 I'm relatively new at this, would you mind pointing me out further?